Strip any trailing slashes when reading the default project path

peterdang1502 opened the pull request:
Fixes #99689. Any trailing slashes in default project path will now be removed.

![image](https://github.com/user-attachments/assets/a404f2e0-94eb-4af2-9609-ea7658c74a29)

![image](https://github.com/user-attachments/assets/47203047-1312-4157-8290-98130a255ab4)

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

Calinou reviewed (approved):
Code looks good to me.

Calinou commented:
Edit: This should also be changed here:

https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/editor/project_manager/project_dialog.cpp#L382-L388

peterdang1502 commented:
> Edit: This should also be changed here:
> 
> https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/editor/project_manager/project_dialog.cpp#L353-L357
> 
> And here:
> 
> https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/editor/project_manager/project_dialog.cpp#L382-L388

I think both of these `path` strings actually get cleaned in later `fdialog_project->set_current_dir()` calls

https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/editor/project_manager/project_dialog.cpp#L358-L367

https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/editor/project_manager/project_dialog.cpp#L389-L395

https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/editor/gui/editor_file_dialog.cpp#L1242-L1246

https://github.com/godotengine/godot/blob/0eadbdb5d0709e4e557e52377fa075d3e2f0ad1f/drivers/windows/dir_access_windows.cpp#L151-L154

My only concerns are these `fix_path()` calls aren't called in `Error DirAccessPack::change_dir` and `Error DirAccessJAndroid::change_dir`, but I'm not sure if they're needed there. Let me know if you want to play it safe and just simplify the path in those two spots anyways. @Calinou

akien-mga commented:
For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

peterdang1502 commented:
> For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

Thank you for spotting that, wasn't aware my local git config was still using my old school email. I have updated my config, all of my commits from now on will be linked to my account.

akien-mga commented:
> > For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.
> 
> Thank you for spotting that, wasn't aware my local git config was still using my old school email. I have updated my config, all of my commits from now on will be linked to my account.

If you want to also fix the identity for the current commit, you can do so with:

```
git commit --amend --reset-author
git push --force
```

peterdang1502 committed:
strip trailing slashes in default project path

peterdang1502 commented:
> > > For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.
> > 
> > 
> > Thank you for spotting that, wasn't aware my local git config was still using my old school email. I have updated my config, all of my commits from now on will be linked to my account.
> 
> If you want to also fix the identity for the current commit, you can do so with:
> 
> ```
> git commit --amend --reset-author
> git push --force
> ```

Thank you for the help, commit is now fixed. I tried to fix it when I changed my local config, but wasn't sure if doing `git push --force`` would add another commit to my PR so I figured it was safe to leave as is but it's fixed now!

Repiteo commented:
Thanks! Congratulations on your first merged contribution! 🎉
